packages/dds/matrix/src/test/matrixBenchmarks.ts

```diff
@@ -0,0 +1,855 @@
+	maxBenchmarkDurationSeconds,
+}: ExecutionTimeBenchmarkConfig): Test {
+	return benchmark({
+		type: BenchmarkType.Measurement,
+		title,
+		benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+			let duration: number;
+			do {
+				assert.equal(state.iterationsPerBatch, 1, "Expected exactly one iteration per batch");
+
+				// Create matrix
+				const { matrix, undoRedoStack, cleanUp } = createTestMatrix({
+					matrixSize,
+					initialCellValue,
+				});
+
+				beforeOperation?.(matrix, undoRedoStack);
+
+				// Operation
+				const before = state.timer.now();
+				operation(matrix, undoRedoStack);
+				const after = state.timer.now();
+
+				// Measure
+				duration = state.timer.toSeconds(before, after);
+
+				afterOperation?.(matrix, undoRedoStack);
+
+				// Cleanup
+				cleanUp();
+			} while (state.recordBatch(duration));
+		},
+		minBatchDurationSeconds,
+		maxBenchmarkDurationSeconds,
+	});
+}
+
+/**
+ * {@link runMemoryBenchmark} configuration.
+ */
+type MemoryBenchmarkConfig = MatrixBenchmarkOptions;
+
+/**
+ * Runs a benchmark for measuring the memory usage of operations on a SharedMatrix.
+ */
+function runMemoryBenchmark({
+	title,
+	matrixSize,
+	initialCellValue,
+	beforeOperation,
+	operation,
+	afterOperation,
+}: MemoryBenchmarkConfig): Test {
+	return benchmarkMemory(
+		new (class implements IMemoryTestObject {
+			readonly title = title;
+
+			private matrix: ISharedMatrix | undefined;
+			private undoRedoStack: UndoRedoStackManager | undefined;
+			private cleanUp: (() => void) | undefined;
+
+			async run(): Promise<void> {
+				assert(this.matrix !== undefined, "matrix is not initialized");
+				assert(this.undoRedoStack !== undefined, "undoRedoStack is not initialized");
+				operation(this.matrix, this.undoRedoStack);
+			}
+
+			beforeIteration(): void {
+				const { matrix, undoRedoStack, cleanUp } = createTestMatrix({
+					matrixSize,
+					initialCellValue,
+				});
+				this.matrix = matrix;
+				this.undoRedoStack = undoRedoStack;
+				this.cleanUp = cleanUp;
+
+				beforeOperation?.(this.matrix, this.undoRedoStack);
+			}
+
+			afterIteration(): void {
+				assert(this.matrix !== undefined, "matrix is not initialized");
+				assert(this.undoRedoStack !== undefined, "undoRedoStack is not initialized");
+				assert(this.cleanUp !== undefined, "cleanUp is not initialized");
+
+				afterOperation?.(this.matrix, this.undoRedoStack);
+
+				this.cleanUp();
+				this.undoRedoStack = undefined;
+				this.cleanUp = undefined;
+			}
+		})(),
+	);
+}
+
+type BenchmarkOptions =
+	| (ExecutionTimeBenchmarkConfig & { mode: "execution-time" })
+	| (MemoryBenchmarkConfig & { mode: "memory" });
+
+function runBenchmark(options: BenchmarkOptions): Test {
+	const mode = options.mode;
+	switch (mode) {
+		case "execution-time": {
+			return runExecutionTimeBenchmark(options);
+		}
+		case "memory": {
+			return runMemoryBenchmark(options);
+		}
+		default: {
+			unreachableCase(mode);
+		}
+	}
+}
+
+/**
```

Comment:
Note for reviewers: no functional changes were made to the tests, they were just deduplicated and moved over here (and updated to reduce some test name redundancy / use our newly unified benchmarking infra).